### PR TITLE
Change fluentbit's crd from namesapce level to cluster level

### PR DESCRIPTION
**Pull request by wenchajun**
Since the `fluentbit-operator` was upgraded to the `fluentd-operator`, the original namespaced level of CRD may not be so suitable. So we need to extend the old CRD to the cluster level. As the new CRD is at the cluster level, the namespace for the newly generated secret file is the namespace under `fluentd-operator`. So the following fields need to be added to the YAML file where the `fluentd-operator` is deployed.

```yaml
env:
  - name: MY_POD_NAMESPACE
    valueFrom:
       fieldRef:
         apiVersion: v1
         fieldPath: metadata.namespace

```

Signed-off-by: chengdehao <dehaocheng@kubesphere.io>

### CODE REVIEW

File: controllers/fluentbitconfig_controller.go [hunk 1]
```diff
@@ -102,12 +103,13 @@ func (r *FluentBitConfigReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		if err = r.List(ctx, &parsers, client.InNamespace(req.Namespace), client.MatchingLabelsSelector{Selector: selector}); err != nil {
+		if err = r.List(ctx, &parsers, client.MatchingLabelsSelector{Selector: selector}); err != nil {
 			return ctrl.Result{}, err
 		}
 
+		ns := os.Getenv("MY_POD_NAMESPACE")
```
Comment (zhu733756) [inline controllers/fluentbitconfig_controller.go:110]: MY_POD_NAMESPACE => NAMESPACE will make sense? And k8s doesn't use NAMESPACE as an env value.

**Comment by zhu733756** [comment]
The GVK in manifests also needs a change.

File: config/crd/bases/fluentbit.fluent.io_fluentbits.yaml [hunk 2]
```diff
@@ -13,10 +13,8 @@ spec:
     kind: FluentBit
     listKind: FluentBitList
     plural: fluentbits
```
Comment (wanjunlei) [inline config/crd/bases/fluentbit.fluent.io_fluentbits.yaml:15]: why the short name was deleted?

Comment (wanjunlei) [inline config/crd/bases/fluentbit.fluent.io_fluentbits.yaml:15]: why the short name was deleted?

File: controllers/fluentbitconfig_controller.go [hunk 3]
```diff
@@ -102,12 +103,13 @@ func (r *FluentBitConfigReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		if err = r.List(ctx, &parsers, client.InNamespace(req.Namespace), client.MatchingLabelsSelector{Selector: selector}); err != nil {
+		if err = r.List(ctx, &parsers, client.MatchingLabelsSelector{Selector: selector}); err != nil {
 			return ctrl.Result{}, err
 		}
 
+		ns := os.Getenv("NAMESPACE")
```
Comment (wanjunlei) [inline controllers/fluentbitconfig_controller.go:110]: It needs to support specifying the namespace where the secret is in. If not specified, the defaults namespace is the namespace of the operator.

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: apis/fluentbit/v1alpha2/clusterfilter_types.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ type FilterSpec struct {
 	// Use this option if you want to use the full regex syntax.
 	MatchRegex string `json:"matchRegex,omitempty"`
 	// A set of filter plugins in order.
-	FilterItems []FilterItem `json:"filters,omitempty"`
+	FilterItems []FilterItem `json:"clusterFilters,omitempty"`
```
Comment (benjaminhuo) [inline apis/fluentbit/v1alpha2/clusterfilter_types.go:43]: We needn't change fields except CRD name to cluster like below to make the migration from namespaced CR to cluster CR easier

filters -> clusterFilters

Comment (wenchajun) [reply apis/fluentbit/v1alpha2/clusterfilter_types.go]: fixed

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
Don't forget the helm need to be updated too @wenchajun

**Comment by wenchajun** [comment]
> Don't forget the helm need to be updated too @wenchajun

Ok, I will update and integrate fluentd's helm

**Pull request merged**